src/js/BookReader.js

```diff
@@ -2985,13 +3018,34 @@ BookReader.prototype.fragmentFromParams = function(params) {
   }
 
   // search
-  if (params.search) {
+  if (params.search && urlMode === 'hash') {
     fragments.push('search', params.search);
   }
 
   return utils.encodeURIComponentPlus(fragments.join(separator)).replace(/%2F/g, '/');
 };
 
+/**
+ * Create, update querystring from the params object
+ *
+ * @param {Object} params
+ * @param {string} currQueryString
+ * @return {string}
+ */
+
+BookReader.prototype.queryStringFromParams = function(
+  params,
+  currQueryString,
+  urlMode = 'hash'
+) {
+  const newParams = new URLSearchParams(currQueryString);
+  if (params.search && urlMode === 'history') {
+    newParams.set('q', params.search)
+  }
+  const result = newParams.toString();
+  return result ? '?' + result : '';
```

Comment:
Q: do you need to append the `?` before the result?

Reply:
Y: URLSearchParams.toString() returns query string with no preceding '?'

Reply:
yah. i saw the string interpolation later in my review. ok 👍